src/cluster_undersample/cluster_undersample.jl

```diff
@@ -0,0 +1,207 @@
+"""
+Assuming that all the observations in the observation matrix X belong to the same class, undersample
+the observations using clustering undersampling
+
+# Arguments
+- `X`: A matrix where each column is an observation of floats
+- `n`: Number of observations in the undersampled dataset
+- `mode::AbstractString="nearest`: If `center` then the undersampled data will consist of the centriods of 
+    each cluster found. Meanwhile, if `nearest` then it will consist of the nearest neighbor of each centroid.
+- `maxiter::Integer=100`: Maximum number of iterations to run K-means
+- `rng::Integer=42`: Random number generator seed
+# Returns
+- `Xnew`: A matrix containing the undersampled observations
+"""
+function cluster_undersample_per_class(
+    X::AbstractMatrix{<:Real},
+    n::Integer;
+    mode::AbstractString = "nearest",
+    maxiter::Integer = 100,
+    rng::Integer = 42,
+)
+    (mode in ["center", "nearest"]) ||
+        throw(ArgumentError("mode must be either 'center' or 'nearest'"))
+    # to undersample down to n points find k=n clusters
+    seed!(rng)               # kmeans offers no better way :(
```

Comment:
> kmeans offers no better way :(

I am not so sure. Perhaps you are confusing *cluster* seeding with RNG seeding. I read in the Clustering.jl docs that users specify the AbstractRNG used in a cluster seeding algorithm using the `initseeds!` method: https://github.com/JuliaStats/Clustering.jl/blob/23bded0cc8f9c4b14e4b031ca939569121183856/src/seeding.jl#L123
No?

Perhaps we replace `rng` argument with `init` for the cluster seeding algorithm and point to the [Clustering.jl documentation](https://juliastats.org/Clustering.jl/stable/kmeans.html#Clustering.kmeans) on options.

Reply:
> I read in the Clustering.jl docs that users specify the AbstractRNG used in a cluster seeding algorithm using the initseeds! method

@ablaom where does it say that in the [docs] (https://juliastats.org/Clustering.jl/stable/init.html#Clustering.initseeds!)? To me it seems that they don't even expose that `rng` argument in the code you linked.

> Perhaps we replace rng argument with init for the cluster seeding algorithm and point to the [Clustering.jl documentation](https://juliastats.org/Clustering.jl/stable/kmeans.html#Clustering.kmeans) on options.

I couldn't quite catch what you meant here. You mean expore the arguments of `initseeds` in the function and then use them in `initseeds`?

Reply:
Okay. Problem solved, after searching through PRs someone did add the feature of specifying an abstract RNG as shown in the [source](https://github.com/JuliaStats/Clustering.jl/blob/master/src/kmeans.jl). For some reason it's not documented.

Reply:
Ahh. I didn't notice the docs were outdated. Good work sorting that out.